protocols/rendezvous/src/client.rs

```diff
@@ -218,9 +224,18 @@ impl NetworkBehaviour for Behaviour {
     }
 
     fn on_swarm_event(&mut self, event: FromSwarm<Self::ConnectionHandler>) {
-        self.external_addresses.on_swarm_event(&event);
+        let changed = self.external_addresses.on_swarm_event(&event);
 
         self.inner.on_swarm_event(event);
+
+        if changed && self.external_addresses.iter().count() > 0 {
+            let registered = self.registered_namespaces.clone();
```

Comment:
Could probably replace the map with an empty one instead of cloning here and allow it to repopulate on a successful registration 

```suggestion
            let registered = std::mem::take(&mut self.registered_namespaces);
```

thoughts?

Reply:
Actually, as I think about it, it might not be best to do since there could be a chance that there is a change in external addresses before the rendezvous node ever response.